src/main/java/uk/nhs/prm/repo/ehrtransferservice/services/ConversationActivityService.java

```diff
@@ -0,0 +1,36 @@
+package uk.nhs.prm.repo.ehrtransferservice.services;
+
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public final class ConversationActivityService {
+    private final Map<UUID, Instant> conversations;
+
+    public ConversationActivityService() {
+        this.conversations = new ConcurrentHashMap<>();
+    }
+
+    public void captureConversationActivityTimestamp(UUID inboundConversationId) {
+        conversations.put(inboundConversationId, Instant.now());
+    }
+
+    public void removeConversationActivityTimestamp(UUID inboundConversationId) {
+        conversations.remove(inboundConversationId);
+    }
+
+    public boolean isConversationActive(UUID inboundConversationId) {
```

Comment:
It is a separate method because in the event of a timeout, we want to call `transferService.updateConversationTransferStatus(inboundConversationId, INBOUND_TIMEOUT);`

If we couple together the timeout & terminating status checks,  we'd have to make the conversationActivityService throw a TimeoutExceededException. We'd have to update RepoIncomingService.waitForConversationToComplete() to wrap it with a Try/Catch block and then rethrow the exception to break the flow further up the chain.

I think the logic doesn't end up any simpler myself, I think it makes more sense to call the two in the Service class and handle any exception throwing there rather than coupling both checks in the conversationActivityService